src/network.c

```diff
@@ -16,8 +16,8 @@ typedef struct {
 	FILE *dl;
 	size_t dl_size;
 
-	size_t pos;
-	size_t limit;
+	off_t pos;
+	off_t limit;
```

Comment:
Why do you used off_t here (instead of goffset or curl_off_t)? And why not for dl_size as well?

Reply:
Mainly because we had size_t before. For `limit` which is passed to `curl_easy_setopt()` probably curl_off_t is the better one. Than we could use this for pos, too, to keep it consistent.

Reply:
`dl_size` is only passed to `open_memstream()` which accepts a `size_t` argument...

Reply:
maybe the right point to drop `download_mem()` which is not used anywhere else than in tests?

Reply:
> maybe the right point to drop `download_mem()` which is not used anywhere else than in tests?

Probably yes, well likely need more control for downloading to memory when/if we implement dm-verity-based bundles, but that might be separate anyway.

Reply:
I've now pushed a version with `curl_off_t` and `download_mem()` removed. The second force push is only a rebase onto current master.